helix-term/src/commands.rs

```diff
@@ -1176,25 +1176,57 @@ fn goto_file_impl(cx: &mut Context, action: Action) {
     let primary = selections.primary();
     // Checks whether there is only one selection with a width of 1
     if selections.len() == 1 && primary.len() == 1 {
-        let count = cx.count();
-        let text_slice = text.slice(..);
-        // In this case it selects the WORD under the cursor
-        let current_word = textobject::textobject_word(
-            text_slice,
-            primary,
-            textobject::TextObject::Inside,
-            count,
-            true,
-        );
-        // Trims some surrounding chars so that the actual file is opened.
-        let surrounding_chars: &[_] = &['\'', '"', '(', ')'];
         paths.clear();
-        paths.push(
-            current_word
-                .fragment(text_slice)
-                .trim_matches(surrounding_chars)
-                .to_string(),
+
+        let is_valid_path_char = |c: &char| {
+            let valid_chars: &[char] = if cfg!(target_os = "windows") {
+                &[
+                    '@', '/', '\\', '.', '-', '_', '+', '#', '$', '%', '{', '}', '[', ']', ':',
+                    '!', '~', '=',
+                ]
+            } else {
+                &['@', '/', '.', '-', '_', '+', '#', '$', '%', '~', '=']
+            };
+            valid_chars.contains(c) || c.is_alphabetic()
+        };
+
+        let start_pos = {
+            let cursor_pos = primary.cursor(text.slice(..));
+            let pre_cursor_pos = cursor_pos.saturating_sub(1);
+            let post_cursor_pos = cursor_pos.saturating_add(1);
+
+            if is_valid_path_char(&text.char(cursor_pos)) {
+                cursor_pos
+            } else if is_valid_path_char(&text.char(pre_cursor_pos)) {
+                pre_cursor_pos
+            } else {
+                post_cursor_pos
+            }
+        };
+
+        let prefix_len = text
+            .chars_at(start_pos)
+            .clone()
```

Comment:
I think this clone is unnecessary, no? I'm surprised that clippy doesn't catch it

Reply:
good catch. Indeed, removing the clone doesn't create an error.